### ISSUE TITLE
"reflect-metadata shim is required when using class decorators" in ngc

### ISSUE THREAD

**Issue by alexeagle** [angular/angular#10857]
ngc uses only `.metadata.json` files and the `StaticReflector` so binaries that run it should not be required to load `reflect_decorators/Reflect.js`.

Currently you get the error in the subject.

**Comment by Delagen** [comment]
Yes. This error occured with latest RC.7 release. When using ngc
My error is
```
undefined, resolving symbol ApplicationModule in .../node_modules/@angular/core/src/application_module.d.ts
```
but when I see in code 
```
 try {
return simplify(value);
}
catch (e) {
throw new Error(e.message + ", resolving symbol " + context.name + " in " + context.filePath);
}
```
this throw error

**Comment by alexeagle** [comment]
@Delagen that's not the same problem as described in this issue. Could you file a new issue with a more complete reproduction of your error?

**Comment by Delagen** [comment]
I simply run ngc on my project that was working fin in RC.6, but now fails with this error on i18n, somewhere in angular/core

**Comment by alexeagle** [comment]
Please file a separate issue, your problem is not related to reflect-metadata.

**Comment by pkozlowski-opensource** [comment]
@alexeagle @chuckjaz is this still the issue?

**Comment by alexeagle** [comment]
It's still *an* issue. we currently have workarounds like https://github.com/angular/angular/blob/master/modules/%40angular/compiler-cli/src/main.ts#L12
and a bunch of similar things in g3 - but we never want to reflect on the TS-generated metadata.

**Comment by alexeagle** [comment]
Was fixed by f4f6f4b4d89c1bfe9112620facb8ee33b2a9ca68
